scenes/game_elements/props/filling_barrel/components/filling_barrel.gd

```diff
@@ -6,8 +6,30 @@ extends StaticBody2D
 
 signal completed
 
-const NEEDED: int = 3
-const DEFAULT_TEXTURE: Texture2D = preload("uid://b2nmajpf8dlh")
+const DEFAULT_SPRITE_FRAMES: SpriteFrames = preload(
+	"res://scenes/game_elements/props/filling_barrel/components/filling_barrel_sprite_frames.tres"
+)
+const FILLING_NAME_ANIMATION: StringName = &"filling"
+
+## Determines how the FillingBarrel looks and also dictate the [member needed_amount]
+## that is needed to fill the barrel.[br]
+## It is required that the [member sprite_frames] have an animation called filling.
+## [br]The barrel default sprite will be the first frame of that animation, and each time
+## the barrel is filled, it will advance one frame.
+@export var sprite_frames: SpriteFrames = DEFAULT_SPRITE_FRAMES:
+	set = _set_sprite_frames
+
+## Read only property that indicates the amount of times the barrel needs to be filled.[br]
+## That amount is calculated as the amount of frames in the filling animation of
+## [member sprite_frames].[br]
+## When the barrel is filled that many times, it emits [signal completed].
+@export_custom(
+	PROPERTY_HINT_NONE,
+	"",
+	PROPERTY_USAGE_READ_ONLY | PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_NO_INSTANCE_STATE
+)
+var needed_amount: int:
+	get = _needed
```

Comment:
I find a bit tricky that the amount depends on the amount of frames. A learner should be able to set the amount number directly in the inspector. Ideally the frames would match, but if not they should be mapped to the current amount. There is a `&"fill"` squash & stretch animation anyways which will indicate the player that the amount has changed. I havent tested but this should do the trick:

```
var total_frames: int = animated_sprite_2d.sprite_frames.get_frame_count(animated_sprite_2d.animation)
animated_sprite_2d.frame = floor((_amount / needed_amount) * total_frames)
```

What do you think?

Reply:
yes, I think that works too!, I like that it lets a learner change the current barrels to make them easier without having to deal with textures or animations. I think it could look a bit weird if the needed amount is > the frame count, but it's true that we have the fill animation so it won't look like nothing happened.